src/rdf4cpp/rdf/writer/BufWriter.hpp

```diff
@@ -181,12 +181,12 @@ struct StringWriter : BufWriterBase<StringWriter, StringBuffer> {
         return buffer();
     }
 
-    static void flush_impl(Buffer &buffer, Cursor &cursor) noexcept {
-        auto const bytes_written = cursor.data() - buffer.data();
+    static void flush_impl(Buffer &buffer, Cursor &cursor, size_t additional_cap) noexcept {
+        auto const bytes_filled = cursor.data() - buffer.data();
 
-        buffer.resize(buffer.size() * 2);
-        cursor.repoint(buffer.data() + bytes_written,
-                       cursor.size() + buffer.size());
+        buffer.resize(std::bit_ceil(buffer.size() + additional_cap));
+        cursor.repoint(buffer.data() + bytes_filled,
+                       buffer.size() - bytes_filled);
```

Comment:
change looks sensible. Growth is now controlled by the data not by exponential growth.

Reply:
still exponential (just sped up) because of `bit_ceil`